### PR TITLE
Wip sophia

### PR DESCRIPTION
**Pull request by sophiafe**
- ML model performance is now as expected when trained on full datasets (with or without limited test set size)
- DL models only tested in debug mode
- added normalized AUPRC (take positive sample fraction into account) and mcc as another good metric for imbalanced datasets
- using python module 3.11.6 for Euler

### CODE REVIEW

**Comment by sophiafe** [comment]
- improved XGBoost and LGBM early_stopping logic to use the validation set
